pkg/runtime/function2/concat.go

```diff
@@ -44,12 +46,19 @@ type concatFunc struct{}
 
 // Apply implements proto.Func.
 func (c concatFunc) Apply(ctx context.Context, inputs ...proto.Valuer) (proto.Value, error) {
-	var sb strings.Builder
+	var (
+		sb       strings.Builder
+		NullType = reflect.TypeOf(ast.Null{}).String()
+	)
 	for _, it := range inputs {
 		val, err := it.Value(ctx)
 		if err != nil {
 			return nil, errors.WithStack(err)
 		}
+		// according to the doc, returns NULL if any argument is NULL.
+		if strings.Compare(reflect.TypeOf(val).String(), NullType) == 0 {
```

Comment:
Use `_,isNull := val.(ast.Null)` if you want to check whether the target is null.

Reply:
done